client/sessiondataset.go

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package client
+
+import "github.com/apache/iotdb-client-go/rpc"
+
+type SessionDataSet struct {
+	ioTDBRpcDataSet *IoTDBRpcDataSet
+}
```

Comment:
Actually, we may only need one DataSet in go-client. For java, we need to support two kinds of dataset which are JDBCDataSet and SessionDataset, so we put the common part into IoTDBRpcDataSet. However, in go-client, we only need to support one.

Reply:
Thank you for your suggestion, IoTDBRpcDataSet is only used in the package, I think we can keep the current structure, maybe in the future we can also adapt to the golang's sql driver.